library/core/src/ptr/mod.rs

```diff
@@ -563,7 +565,9 @@ pub const fn null<T: ?Sized + Thin>() -> *const T {
 #[rustc_allow_const_fn_unstable(ptr_metadata)]
 #[rustc_diagnostic_item = "ptr_null_mut"]
 pub const fn null_mut<T: ?Sized + Thin>() -> *mut T {
-    from_raw_parts_mut(invalid_mut(0), ())
+    // Use transmute instead casting so that miri knows that the pointer is invalid.
+    // SAFETY: every usize is a valid pointer (on all current platforms).
```

Comment:
```suggestion
    // Use transmute instead of casting so that Miri knows that the pointer is invalid.
    // SAFETY: on all current platforms, usize and pointers have the same layout, and the validity invariant of pointers is the same as that of integers
```

don't need to r- unless you want to, could do as a follow-up (or just ignore^^)

Reply:
Yeah, that's much better, thanks! 😄